sporkfish/searcher.py

```diff
@@ -348,6 +350,59 @@ def _negamax_lazy_smp(
             else:
                 continue  # Continue the loop if no result is ready yet
 
+    # Aspiriation windows
+    # Assumption (*): we expect the search score from depth n should be somewhat similar depth n + 1.
+    # Method: set alpha = prev_depth_score - constant, beta = prev_depth_score + constant.
+    # Do a search with narrower window [alpha, beta] as opposed to [-inf, inf].
+    # This should cause cutoffs (if they do happen) to happen faster.
+    # If the score falls within the alpha/beta bounds, our assumption (*) holds - so take that score.
+    # If the score falls outside the alpha/beta windows, the score changed too much
+    # i.e. our assumption (*) is false - so we do a full search to be safe.
+    def _aspiration_windows_search(
+        self,
+        search_func: Callable[[Board, int, float, float], Tuple[float, chess.Move]],
```

Comment:
Wasn't immediately clear to me what the `search_func` mean. I can see it is calling, e.g. `negamax_sp`.

I think would be clearer when I refactored this searcher class a bit.

Reply:
Yes - hopefully abit clearer in MiniMaxVariant, which can directly call the search function.